db/crud.go

```diff
@@ -1754,6 +1754,12 @@ func (db *Database) updateAndReturnDoc(docid string, allowImport bool, expiry ui
 				doc.inlineSyncData = true
 			}
 
+			userXattrs := currentUserXattr
+			if existingDoc != nil {
+				userXattrs = existingDoc.UserXattr
+			}
+			doc.RawUserXattr = userXattrs
```

Comment:
It looks like you could skip the temporary variable userXattrs and just assign straight to doc.RawUserXattr on line 1757 and 1759.